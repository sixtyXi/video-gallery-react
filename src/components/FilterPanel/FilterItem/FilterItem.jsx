// @flow

import React, { Fragment } from 'react';
import classNames from 'classnames/bind';

import styles from './FilterItem.scss';

const cx = classNames.bind(styles);

export type FilterType = 'btn' | 'label';

type Props = {
  name: string,
  handleChange: Function,
  filterValue: string,
  filterTitle: string,
  checked: boolean,
  type: FilterType
};

export const FilterItem = ({
  name,
  handleChange,
  filterValue,
  filterTitle,
  checked,
  type
}: Props) => {
  const className = cx({
    btn: type === 'btn',
    label: type === 'label'
  });

  return (
    <Fragment>
      <input
        className={styles.filterInput}
        type="radio"
        name={name}
        onChange={handleChange}
        id={filterValue}
        value={filterValue}
        checked={checked}
      />
      <label className={className} htmlFor={filterValue}>
        {filterTitle}
      </label>
    </Fragment>
  );
};

FilterItem.defaultProps = {
  type: 'btn',
  checked: false
};
