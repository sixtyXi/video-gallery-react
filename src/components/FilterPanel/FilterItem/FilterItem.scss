@import '../../../shared/_colors.scss';
@import '../../../shared/_typography.scss';

.filterInput {
  display: none;
}

label {
  cursor: pointer;
}

.btn {
  margin-left: 10px;
  padding: 0 5px;
  background: $secondary;

  .filterInput:checked + & {
    background: $accent;
  }
}

.label {
  margin-left: 15px;

  .filterInput:checked + & {
    color: $accent;
  }
}
