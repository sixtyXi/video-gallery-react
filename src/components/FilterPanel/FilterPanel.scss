.filterWrapper {
  display: flex;
  align-items: center;
}
