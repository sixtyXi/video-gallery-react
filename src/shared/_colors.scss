$accent: #fd4d61;
$secondary: #414141;
