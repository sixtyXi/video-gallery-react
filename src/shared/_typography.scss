$text-light: white;
$size-m: 14px;
$size-s: 12px;
$size-xs: 10px;
$size-l: 16px;
