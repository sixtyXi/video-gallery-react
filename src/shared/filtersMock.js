export const SORT_FILTERS = [
  {
    value: 'release_date',
    title: 'release date'
  },
  {
    value: 'vote_average',
    title: 'rating'
  }
];

export const SEARCH_FILTERS = [
  {
    value: 'title',
    title: 'title'
  },
  {
    value: 'genres',
    title: 'genre'
  }
];

export const DEFAULT_FILTERS = {
  searchBy: SEARCH_FILTERS[0].value,
  sortBy: SORT_FILTERS[0].value,
  search: ''
};
