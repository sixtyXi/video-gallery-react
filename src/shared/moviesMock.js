export const MOVIES = [
  {
    id: 680,
    title: 'Pulp Fiction',
    tagline: "Just because you are a character doesn't mean you have character.",
    vote_average: 8.3,
    vote_count: 10910,
    release_date: '1994-09-10',
    poster_path: 'https://image.tmdb.org/t/p/w500/dM2w364MScsjFf8pfMbaWUcWrR.jpg',
    overview:
      "A burger-loving hit man, his philosophical partner, a drug-addled gangster's moll and a washed-up boxer converge in this sprawling, comedic crime caper. Their adventures unfurl in three stories that ingeniously trip back and forth in time.",
    budget: 8000000,
    revenue: 213928762,
    genres: ['Thriller', 'Crime'],
    runtime: 154
  },
  {
    id: 2,
    title: 'Kill Bill: Vol. 1',
    tagline: 'Oscar-winning Movies',
    vote_average: 4.5,
    release_date: '2003',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/71XwrZLWOBL._SY450_.jpg',
    overview: `From groundbreaking writer and director Quentin Tarantino KILL BILL VOLUME 1 stars Uma Thurman, Lucy Liu, and Vivica A. Fox in an epic vendetta! Four years after taking a bullet in the head at her own wedding, The Bride emerges from a coma and decides it's time for payback! Having been gunned down by her former boss and his deadly squad of international assassins, it's a kill-or- be- killed fight.`,
    runtime: 150,
    genres: ['action', 'adventure']
  },
  {
    id: 3,
    title: 'Four rooms',
    tagline: 'Oscar-winning Movies',
    vote_average: 3,
    release_date: '1995',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/51qcn-rYWqL._SY450_.jpg',
    overview: `Don't miss the fun in this hilariously sexy comedy that has Antonio Banderas (Once Upon a Time in Mexico), Madonna (Swept Away), and a sizzling all-star cast checking in for laughs! It's Ted the Bellhop's (Tim Roth, Pulp Fiction) first night on the job...and the hotel's very unusual guests are about to place him in some outrageous predicaments.`,
    runtime: 97,
    genres: ['comedies']
  },
  {
    id: 4,
    title: 'Kill Bill: Vol. 1',
    tagline: 'Oscar-winning Movies',
    vote_average: 4.5,
    release_date: '2003',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/71XwrZLWOBL._SY450_.jpg',
    overview: `From groundbreaking writer and director Quentin Tarantino KILL BILL VOLUME 1 stars Uma Thurman, Lucy Liu, and Vivica A. Fox in an epic vendetta! Four years after taking a bullet in the head at her own wedding, The Bride emerges from a coma and decides it's time for payback! Having been gunned down by her former boss and his deadly squad of international assassins, it's a kill-or- be- killed fight.`,
    runtime: 150,
    genres: ['action', 'adventure']
  },
  {
    id: 5,
    title: 'Four rooms',
    tagline: 'Oscar-winning Movies',
    vote_average: 3,
    release_date: '1995',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/51qcn-rYWqL._SY450_.jpg',
    overview: `Don't miss the fun in this hilariously sexy comedy that has Antonio Banderas (Once Upon a Time in Mexico), Madonna (Swept Away), and a sizzling all-star cast checking in for laughs! It's Ted the Bellhop's (Tim Roth, Pulp Fiction) first night on the job...and the hotel's very unusual guests are about to place him in some outrageous predicaments.`,
    runtime: 97,
    genres: ['comedies']
  },
  {
    id: 6,
    title: 'Kill Bill: Vol. 1',
    tagline: 'Oscar-winning Movies',
    vote_average: 4.5,
    release_date: '2003',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/71XwrZLWOBL._SY450_.jpg',
    overview: `From groundbreaking writer and director Quentin Tarantino KILL BILL VOLUME 1 stars Uma Thurman, Lucy Liu, and Vivica A. Fox in an epic vendetta! Four years after taking a bullet in the head at her own wedding, The Bride emerges from a coma and decides it's time for payback! Having been gunned down by her former boss and his deadly squad of international assassins, it's a kill-or- be- killed fight.`,
    runtime: 150,
    genres: ['action', 'adventure']
  },
  {
    id: 7,
    title: 'Four rooms',
    tagline: 'Oscar-winning Movies',
    vote_average: 3,
    release_date: '1995',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/51qcn-rYWqL._SY450_.jpg',
    overview: `Don't miss the fun in this hilariously sexy comedy that has Antonio Banderas (Once Upon a Time in Mexico), Madonna (Swept Away), and a sizzling all-star cast checking in for laughs! It's Ted the Bellhop's (Tim Roth, Pulp Fiction) first night on the job...and the hotel's very unusual guests are about to place him in some outrageous predicaments.`,
    runtime: 97,
    genres: ['comedies']
  }
];
