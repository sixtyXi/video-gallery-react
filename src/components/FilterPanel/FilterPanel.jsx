// @flow
import * as React from 'react';

import styles from './FilterPanel.scss';
import { FilterItem } from './FilterItem/FilterItem';
import type { FilterType } from './FilterItem/FilterItem';
import type { Filter } from '../../shared/types';

type Props = {
  filters: Array<Filter>,
  activeFilter: string,
  name: string,
  handleChange: Function,
  type: FilterType,
  title: string
};

export const FilterPanel = ({
  filters,
  activeFilter,
  name,
  handleChange,
  type,
  title
}: Props): React.Node => {
  const filterItems = filters.map(filter => {
    const isChecked: boolean = activeFilter === filter.value;

    return (
      <FilterItem
        key={filter.value}
        name={name}
        handleChange={handleChange}
        filterValue={filter.value}
        filterTitle={filter.title}
        type={type}
        checked={isChecked}
      />
    );
  });

  return filterItems.length ? (
    <div className={styles.filterWrapper}>
      <p className={styles.filterTitle}>{title}</p>
      {filterItems}
    </div>
  ) : null;
};

FilterPanel.defaultProps = {
  filters: [],
  activeFilter: '',
  type: 'btn',
  title: 'Search by',
  handleChange: () => {},
  name: 'filterValue'
};
