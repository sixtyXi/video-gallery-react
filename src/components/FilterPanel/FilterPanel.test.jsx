import React from 'react';
import { shallow } from 'enzyme';

import { FilterPanel } from './FilterPanel';
import { FilterItem } from './FilterItem/FilterItem';
import { SORT_FILTERS } from '../../shared/filtersMock';

describe('FilterPanel', () => {
  const title = 'Sort by';

  test('renders correctly', () => {
    const wrapper = shallow(<FilterPanel filters={SORT_FILTERS} title={title} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('not renders if filters are empty', () => {
    const wrapper = shallow(<FilterPanel title={title} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('renders title', () => {
    const wrapper = shallow(<FilterPanel filters={SORT_FILTERS} title={title} />);
    const titleElem = wrapper.find('.filterTitle');

    expect(titleElem.text()).toBe(title);
  });

  test('renders two components - *FilterItem*', () => {
    const handleChange = jest.fn();

    const wrapper = shallow(<FilterPanel filters={SORT_FILTERS} handleChange={handleChange} />);

    expect(wrapper.find(FilterItem).length).toBe(2);
  });
});
