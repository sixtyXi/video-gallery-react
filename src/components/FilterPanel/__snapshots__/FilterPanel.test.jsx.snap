// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterPanel not renders if filters are empty 1`] = `""`;

exports[`FilterPanel renders correctly 1`] = `
<div
  className="filterWrapper"
>
  <p
    className="filterTitle"
  >
    Sort by
  </p>
  <FilterItem
    checked={false}
    filterTitle="release date"
    filterValue="release_date"
    handleChange={[Function]}
    key="release_date"
    name="filterValue"
    type="btn"
  />
  <FilterItem
    checked={false}
    filterTitle="rating"
    filterValue="vote_average"
    handleChange={[Function]}
    key="vote_average"
    name="filterValue"
    type="btn"
  />
</div>
`;
